--- libfabric-1.4.1/prov/gni/configure.m4.orig	2017-04-15 18:30:07.926608137 +0200
+++ libfabric-1.4.1/prov/gni/configure.m4	2017-04-15 18:33:35.529939101 +0200
@@ -71,10 +71,7 @@
                       [AC_DEFINE_UNQUOTED([HAVE_XPMEM], [0], [Define to 1 if xpmem available])
                       ])
 
-               gni_path_to_gni_pub=${CRAY_GNI_HEADERS_CFLAGS:2}
-dnl looks like we need to get rid of some white space
-               gni_path_to_gni_pub=${gni_path_to_gni_pub%?}/gni_pub.h
-               gni_path_to_gni_pub=${CRAY_GNI_HEADERS_CFLAGS:2}
+               gni_path_to_gni_pub=${CRAY_GNI_HEADERS_CFLAGS#-I}
 dnl looks like we need to get rid of some white space
                gni_path_to_gni_pub=${gni_path_to_gni_pub%?}/gni_pub.h
 
